src/aws/mod.rs

```diff
@@ -122,6 +128,8 @@ pub async fn create_client<T: ClientBuilder>(
     };
 
     config_builder = T::with_sleep_impl(config_builder, Arc::new(TokioSleep));
+    // we disable retries because we have our own retry layer wired together with ARC to backoff
+    config_builder = T::with_retry_config(config_builder, RetryConfig::new().with_max_attempts(1));
```

Comment:
I think this means that there is 1 _retry_ attempt. It's not really clear from the documentation though. I'd switch to `retryConfig::disabled()` which explicitly disables retries.

Reply:
Ah good call, I actually set it to 1 based on the docs which say it has to be at least 1 (to have one attempt). Looking at `RetryConfig::disabled()` this appears to be what it does:

https://docs.rs/aws-smithy-types/0.41.0/src/aws_smithy_types/retry.rs.html#217

It's more explicit though, for sure. I'll update.

Reply:
[a353a52](https://github.com/vectordotdev/vector/pull/12611/commits/a353a5215027e498a936ba28e3c350b884ea3b82)